sequencer/sequencer.go

```diff
@@ -358,6 +359,26 @@ func (s *Sequencer) shouldSendSequences(ctx context.Context) (bool, bool) {
 	}
 
 	if err != nil {
+		// while estimating gas a new block is not created and the POE SC may return
+		// an error regarding timestamp verification, this must handled
+		if strings.Contains(err.Error(), errTimestampMustBeInsideRange) {
+			// query the sc about the value of its lastTimestamp variable
+			lastTimestamp, err := s.etherman.GetLastTimestamp()
+			if err != nil {
+				log.Errorf("failed to estimate gas for sequence batches, err: %v", err)
+				return false, false
+			}
+			// check POE SC lastTimestamp against sequences' one
+			for _, seq := range s.closedSequences {
+				if seq.Timestamp < int64(lastTimestamp) {
+					panic("sequence timestamp is < POE SC lastTimestamp")
+				}
```

Comment:
update the lasTimestamp to use the current sequence timestamp, to better reflect the SC require

```suggestion
				}
				lastTimeStamp = seq.Timestamp
```

Reply:
I don't get this. lastTimestamp is not used later in the code.

Reply:
The idea is that in the next iteration, `if seq.Timestamp < int64(lastTimestamp) {` lastTimeStamp should be the timpestamp of the previous item in the sequences

Reply:
Ok, now I see. During SC execution the lastTimestamp will be updated with the previous (known to be correct) one.